dss/api/subscriptions.py

```diff
@@ -94,22 +85,32 @@ def put(json_request_body: dict, replica: str):
     # john@example.com would show up because elasticsearch matched example w/ example.
     # By including "index": "not_analyzed", Elasticsearch leaves all owner inputs alone.
     index_name = Config.get_es_index_name(ESIndexType.subscriptions, Replica[replica])
-    #  TODO (tsmith): get all indexes that use current alais
-    #  TODO (tsmith): try to subscribe query to each of the indexes.
-    #  TODO (tsmith): error if no queries are indexed.
     get_elasticsearch_subscription_index(es_client, index_name, logger, index_mapping)
 
-    try:
-        percolate_registration = _register_percolate(es_client, uuid, es_query, replica)
-        logger.debug(f"Percolate query registration succeeded:\n{percolate_registration}")
-    except ElasticsearchException as ex:
+    #  get all indexes that use current alias
+    alias_name = Config.get_es_alias_name(ESIndexType.docs, Replica[replica])
+    doc_indexes = [i['i'] for i in es_client.cat.aliases(name=alias_name, h=['a', 'i'], format='json')]
+    #  try to subscribe query to each of the indexes.
+    subscribed_indexes = []
+    for doc_index in doc_indexes:
+        try:
+
+            percolate_registration = _register_percolate(es_client, doc_index, uuid, es_query, replica)
+            logger.debug(f"Percolate query registration succeeded:\n{percolate_registration}")
+            subscribed_indexes.append(doc_index)
+        except ElasticsearchException as ex:
+            last_ex = ex
+
+    # error if no queries are successfully indexed.
+    if len(subscribed_indexes) == 0:
         logger.critical("%s", f"Percolate query registration failed: owner: {owner}, uuid: {uuid}, "
-                              f"replica: {replica}, es_query: {es_query}, Exception: {ex}")
+                              f"replica: {replica}, es_query: {es_query}, Exception: {Exception(last_ex)}")
         raise DSSException(requests.codes.internal_server_error,
                            "elasticsearch_error",
-                           "Unable to register elasticsearch percolate query!")
+                           "Unable to register elasticsearch percolate query!") from Exception(last_ex)
```

Comment:
This is wrapping it again. Might want to extract a variable as in  `last_ex = Exception(last_ex)` above.